pkg/pool-manager/pool_test.go

```diff
@@ -413,5 +414,53 @@ var _ = Describe("Pool", func() {
 			Expect(newPod.Annotations[networksAnnotation]).To(Equal(afterAllocationAnnotation[networksAnnotation]))
 		})
 	})
+	Describe("Multus Network Annotations API Tests", func() {
+		It("should allocate requested mac-address with single ip-address request in network annotation", func() {
+			poolManager := createPoolManager("02:00:00:00:00:00", "02:00:00:00:00:01", &vmConfigMap)
+			Expect(poolManager).ToNot(Equal(nil), "failed to create pool-manager")
 
+			expectedMacRequestWithIpAnnotation := `[{"name":"ovs-conf","namespace":"default","ips":["10.10.0.1"],"mac":"02:00:00:00:00:00","cni-args":null}]`
+			newPod := samplePod
+			newPod.Name = "testPod"
+			newPod.Annotations = map[string]string{networksAnnotation: `[{"name":"ovs-conf","namespace":"default","ips":["10.10.0.1"]}]`}
+
+			err := poolManager.AllocatePodMac(&newPod)
+			Expect(err).ToNot(HaveOccurred(),"failed to handle allocate-pod-mac request with ip-address")
+			Expect(newPod.Annotations[networksAnnotation]).To(Equal(expectedMacRequestWithIpAnnotation),
+				fmt.Sprintf("Pod networtk annotation is not as expected,\n"+
```

Comment:
You have to use Gomega JSON matcher MatchJSON so this check is not affected by json attribute re-order.

The "obtained" and "expected" typical test log has to be printed out by Gomega not by us, the string should be something like "Should annotate pod with IP and MAC"

Reply:
done